### PR TITLE
Fix 647 Support Node 18 as Target

### LINKED ISSUE #647: node 18 support
Are there any plans and/or timeline to support node 18? Thank you!

### PR DESCRIPTION
**Pull request by ronilan**
### Overview

This pull request fixes #647

### Change
- Updated `abi_crosswalk.json` to include node 18 (current) (via script).
- Added test coverage for cases of unsupported/unknown targets.

### CODE REVIEW

**Comment by acalcutt** [comment]
Would love to see this merged for the node 18 support

**Review by axrj — APPROVED** [review]
Thanks!

**Comment by axrj** [comment]
Hey! I noticed CI job is missing for node 18, mind adding that?

**Comment by ronilan** [comment]
> Hey! I noticed CI job is missing for node 18, mind adding that?

@axrj assuming this was referring me, I wrote this PR (and all others leading to #656) last year using two weeks of @solarwindscloud time. I’m no longer on their paycheck (and I think they moved away from this package, @cheempz can confirm).

As said earlier this year in #657, if there is interest from corporate users of the package to fund more active maintenance and/or tackling some of the more prominent open issues, and/or potentially taking it off @mapbox - I a should be technically able.

**Review by cclauss — DISMISSED** [review]
* #709 
* #710

File: lib/util/abi_crosswalk.json [hunk 1]
```diff
@@ -2598,5 +2662,57 @@
...
+    "v8": "9.6"
+  },
+  "18.0.0": {
+    "node_abi": 108,
+    "v8": "10.1"
+  },
+  "18.1.0": {
+    "node_abi": 108,
+    "v8": "10.1"
   }
```
Comment (cclauss) [inline lib/util/abi_crosswalk.json:2717]: Below a ⛔ symbol appears because the GitHub editor wants all text files to end with one and only one blank line.

**Comment by cclauss** [comment]
@ronilan, @ewanharris Would you be willing to collaborate to make a single pull request that supports all current versions of Node.js with successful AppVeyor tests?

@ronilan you could bring in the changes from #709 into this pull request and update support for Node.js up to v21 to `lib/util/abi_crosswalk.json`.

@ewanharris you could help me to add your `patch-package` trick (https://github.com/mapbox/node-pre-gyp/issues/710#issuecomment-2049375155) to the AppVeyor process.

By combining our knowledge we should provide a working example of taking the current codebase and patching in updates to `lib/util/abi_crosswalk.json` and passing tests on all currently supported versions of Node.js.

Thoughts?

**Comment by ewanharris** [comment]
@cclauss the patch-package process is more for consumers of node-pre-gyp to workaround the lack of updates, it allows patching the `abi_crosswalk.json` file after installing node_modules to allow builds to work on newer versions.

I'd be happy to PR an updated `abi_crosswalk.json` file to the branch for this update in @ronilan's repo so that this PR then gets updated with the latest versions once that PR is merged,

**Comment by acalcutt** [comment]
Just an FYI for anyone interested. I have updated [abi_crosswalk.json](https://github.com/acalcutt/node-pre-gyp/blob/master/lib/util/abi_crosswalk.json) to node 21 in my fork of node-pre-gyp at [@acalcutt/node-pre-gyp](https://www.npmjs.com/package/@acalcutt/node-pre-gyp) , which I forked for the node version of maplibre. 

Unfortunately it doesn't seem like anyone is maintaining this here.... since this has been open a while with no interest and now needs further updating.

**Comment by acalcutt** [comment]
I added https://github.com/mapbox/node-pre-gyp/pull/711 to update abi_crosswalk.json to node 21 if anyone is interested in updating the support.

**Comment by acalcutt** [comment]
I added a second PR which I think makes the changes requsted by @cclauss in https://github.com/mapbox/node-pre-gyp/pull/649#issuecomment-2049490190 
 
https://github.com/mapbox/node-pre-gyp/pull/712

**Review by cclauss — APPROVED** [review]

**Pull request merged**